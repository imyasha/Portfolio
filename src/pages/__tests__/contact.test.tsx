import React from 'react';
import 'jest-styled-components';
import '@testing-library/jest-dom';
import * as Gatsby from 'gatsby';
import { render } from '@testing-library/react';
import Contact from '../contact';
import Thanks from '../contact/thanks';

const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery');
useStaticQuery.mockImplementation(() => ({
  site: {
    siteMetadata: {
      description: '',
      title: 'Contact',
      lang: 'en',
      meta: [],
      keywords: [],
    },
  },
}));

test('renders the contact page correctly', () => {
  const { container } = render(<Contact />);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders inputs with associated labels', () => {
  const { getByLabelText } = render(<Contact />);
  const nameInput = getByLabelText(/your name/i);
  expect(nameInput).toHaveAttribute('type', 'text');
  const emailInput = getByLabelText(/your email/i);
  expect(emailInput).toHaveAttribute('type', 'email');
  const message = getByLabelText(/your message/i);
  expect(message).toHaveAttribute('name', 'message');
  expect(message).toBeRequired();
});

test('inputs are required', () => {
  const { getByLabelText } = render(<Contact />);
  const nameInput = getByLabelText(/your name/i);
  expect(nameInput).toBeRequired();
  const emailInput = getByLabelText(/your email/i);
  expect(emailInput).toBeRequired();
  const message = getByLabelText(/your message/i);
  expect(message).toBeRequired();
});

test('renders the thank you page correctly', () => {
  const { container } = render(<Thanks />);
  expect(container.firstChild).toMatchSnapshot();
});
