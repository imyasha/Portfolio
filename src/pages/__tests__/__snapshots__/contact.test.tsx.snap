// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the contact page correctly 1`] = `
.c0 {
  width: 1140px;
  max-width: 94%;
  margin: 0 auto;
}

.c1 label {
  width: 100%;
  font-size: 1.5rem;
}

.c1 input {
  height: 30px;
  width: 93%;
}

.c1 textarea {
  height: 90px;
  width: 93%;
}

.c1 button {
  width: 100%;
  background: #e94e1b;
  color: #fff;
  margin-bottom: 5rem;
}

.c1 input,
.c1 textarea,
.c1 button {
  border-radius: calc(15px + 0.25rem);
  border: none;
  padding: 0.25rem 1rem;
  font-size: 1.25rem;
  font-family: 'brandon-grotesque',sans-serif;
  display: block;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  max-width: 90%;
  width: 500px;
  margin: 1.5rem auto 0;
}

<div
  class="c0"
>
  <h1
    data-testid="headline"
  >
    Get in touch
  </h1>
  <p>
    Wanna talk? Send me an email through the form below or get in touch through my social media profiles.
  </p>
  <form
    action="/contact/thanks/"
    class="c1"
    data-netlify="true"
    data-netlify-honeypot="bot-field"
    method="post"
    name="Contact Form"
  >
    <input
      name="bot-field"
      type="hidden"
    />
    <div
      class="c2"
    >
      <label
        for="name"
      >
        Your Name
        <input
          id="name"
          name="name"
          required=""
          type="text"
        />
      </label>
    </div>
    <div
      class="c2"
    >
      <label
        for="email"
      >
        Your Email
        <input
          id="email"
          name="email"
          required=""
          type="email"
        />
      </label>
    </div>
    <div
      class="c2"
    >
      <label
        for="message"
      >
        Your Message
        <textarea
          id="message"
          name="message"
          required=""
        />
      </label>
    </div>
    <div
      class="c2"
    >
      <button
        type="submit"
      >
        Send Â»
      </button>
    </div>
  </form>
</div>
`;

exports[`renders the thank you page correctly 1`] = `
.c0 {
  width: 1140px;
  max-width: 94%;
  margin: 0 auto;
}

<div
  class="c0"
>
  <h1>
    Thank you for your message
  </h1>
  <p>
    Your message was sent. I'll get back to you as soon as I can.
  </p>
</div>
`;
